source/patterns/@aws-solutions-constructs/core/lib/cloudfront-distribution-defaults.ts

```diff
@@ -18,32 +18,43 @@ import * as origins from 'aws-cdk-lib/aws-cloudfront-origins';
 import * as mediastore from 'aws-cdk-lib/aws-mediastore';
 import * as s3 from 'aws-cdk-lib/aws-s3';
 import * as cdk from 'aws-cdk-lib';
+import {BehaviorOptions} from "aws-cdk-lib/aws-cloudfront/lib/distribution";
 
 export function DefaultCloudFrontWebDistributionForApiGatewayProps(apiEndPoint: api.RestApi,
   loggingBucket: s3.Bucket | undefined,
   setHttpSecurityHeaders: boolean,
-  cfFunction?: cloudfront.IFunction): cloudfront.DistributionProps {
+  cfFunction?: cloudfront.IFunction,
+  responseHeadersPolicy?: cloudfront.ResponseHeadersPolicy,
+): cloudfront.DistributionProps {
 
   const apiEndPointUrlWithoutProtocol = cdk.Fn.select(1, cdk.Fn.split("://", apiEndPoint.url));
   const apiEndPointDomainName = cdk.Fn.select(0, cdk.Fn.split("/", apiEndPointUrlWithoutProtocol));
 
+  let defaultBehavior: BehaviorOptions = {
+    origin: new origins.HttpOrigin(apiEndPointDomainName, {
+      originPath: `/${apiEndPoint.deploymentStage.stageName}`
+    }),
+    ...getFunctionAssociationsProp(setHttpSecurityHeaders, cfFunction),
+    viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
+  };
+  if (responseHeadersPolicy) {
```

Comment:
Of course, that's what I did first.
But that broke existing tests, because having the property with value undefined is technically different from not having the property at all. If I recall correctly, having the property as undefined changed the CloudFormation template output.
I agree that it would work the same, but I'd rather not change external behavior. Your call, do you want me to change that anyway?

Reply:
If that breaks existing tests, then don't do what I suggest. But this would work, correct?

```
if (responseHeadersPolicy) {
    defaultBehavior.responseHeadersPolicy = responseHeadersPolicy;
}
```

This doesn't allocate a new object, so might be more efficient - but my main argument is personal, as someone late in the game to Typescript I find it easier to understand than the spread operator.

Reply:
I agree that the assignment would be more efficient and easier to understand than the spread operator.
Unfortunately assignment is not possible, the properties of defaultBehavior are readonly.
I tried the assignment, too. Error: `Attempt to assign to const or readonly variable `

The only way I see to avoid the spread operator would be to duplicate the code:
```
  const defaultBehavior: BehaviorOptions = responseHeadersPolicy ? {
    origin: new origins.HttpOrigin(apiEndPointDomainName, {
      originPath: `/${apiEndPoint.deploymentStage.stageName}`
    }),
    ...getFunctionAssociationsProp(setHttpSecurityHeaders, cfFunction),
    viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
    responseHeadersPolicy
  } : {
    origin: new origins.HttpOrigin(apiEndPointDomainName, {
      originPath: `/${apiEndPoint.deploymentStage.stageName}`
    }),
    ...getFunctionAssociationsProp(setHttpSecurityHeaders, cfFunction),
    viewerProtocolPolicy: cloudfront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,
  };
```
  
  Do you like this better than my current code?

Reply:
OK, let's go with yours code.